packages/cron/src/jobs/metrics.js

```diff
@@ -0,0 +1,89 @@
+import settle from 'p-settle'
+
+/**
+ * @typedef {import('pg').Client} Client
+ * @typedef {{ name: string, value: number }} Metric
+ */
+
+const COUNT_USERS = 'SELECT COUNT(*) AS total FROM public.user'
+
+const COUNT_UPLOADS = 'SELECT COUNT(*) AS total FROM upload WHERE type = $1'
+
+const COUNT_PINS =
+  'SELECT COUNT(*) AS total FROM pin WHERE service = $1 AND status = $2'
+
+const UPDATE_METRIC = `
+INSERT INTO metric (name, value, updated_at)
+     VALUES ($1, $2, TIMEZONE('utc', NOW()))
+ON CONFLICT (name) DO UPDATE
+        SET value = $2, updated_at = TIMEZONE('utc', NOW())
+`
+
+const UPLOAD_TYPES = ['Car', 'Blob', 'Multipart', 'Remote', 'Nft']
+const PIN_SERVICES = ['IpfsCluster2', 'IpfsCluster', 'Pinata']
+const PIN_STATUSES = ['PinQueued', 'Pinning', 'Pinned', 'PinError']
+
+/**
+ * Calculate metrics and update their current values in the DB.
+ *
+ * @param {{ pg: Client, roPg: Client }} config
+ */
+export async function updateMetrics({ pg, roPg }) {
+  const metrics = await settle([
+    getUsersCount(roPg),
+    ...UPLOAD_TYPES.map((t) => getUploadsCount(roPg, t)),
+    ...PIN_SERVICES.map((svc) =>
+      PIN_STATUSES.map((s) => getPinsCount(roPg, svc, s))
+    ).flat(),
+  ])
+
+  let error
+  for (const promise of metrics) {
+    if (promise.isRejected) {
+      error = error || promise.reason
+      console.error(promise.reason)
+      continue
+    }
+    const metric = promise.value
+    await pg.query(UPDATE_METRIC, [metric.name, metric.value])
+  }
+
+  if (error) throw error
```

Comment:
Good call that we update what we can and only error in the end. We still need to create an issue to integrate these errors with an alert system.

Reply:
I saw this in the pins job https://github.com/nftstorage/nft.storage/blob/20ee0b6241917c92821e4d7b79e19e4ae36ccea1/.github/workflows/cron-pins.yml#L35-L37

Is that what we want @hugomrdias?

Reply:
Oh yes, I think that's it. Not sure if we need any config in the expected opsgenie heartbeat timings